### PR TITLE
h264: added testing for mbPartPredMode and corrected misbehaviour 

### LINKED ISSUE #28: h264: write testing for MbPartPredMode
https://github.com/ausocean/h264decode/blob/c67440a589f2d6f4ade91b1f67c6f80cdbed44db/h264/mbType.go#L90-L163

This function needs testing and also possibly some modification to work correctly. See section 7.4.5 and tables 7-11+ in specs for guidance on expected results.

### PR DESCRIPTION
**Pull request by saxon-milton**
Resolves #28

### CODE REVIEW

File: h264/mbtype_test.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package h264
```
Comment (kortschak) [inline h264/mbtype_test.go:11]: This shouldn't be a commit.

Comment (saxon-milton) [reply h264/mbtype_test.go:11]: Should I squash ?

Comment (kortschak) [reply h264/mbtype_test.go:11]: Yes.

Comment (saxon-milton) [reply h264/mbtype_test.go:11]: squashed.

**Review by kortschak — APPROVED** [review]
Comments on the constants would be nice.

**Pull request merged**